Source/Modules/scilab.cxx

```diff
@@ -254,7 +254,7 @@ class SCILAB:public Language {
 
     // Add Builder footer code and save
     if (generateBuilder) {
-      saveBuilderFile(gatewayName);
+      saveBuilderFile(gatewayLibraryName);
```

Comment:
filename   past to    function saveBuilderFile

saveBuilderFile(gatewayLibraryName);

must be same as that pass to  this line that the appears in swig generated script file 'loader.sce'

   Printf(loaderScript, "  [bOK, ilib] = c_link('%s');\n", gatewayLibraryName); 

So, if you kept to using  'gatewayName' than it would not be loaded by loader.sce because the generated script file using 'gatewayLibraryName'. 

This inconsistency in file name may be cause trouble in Linux system.

Reply:
Take scilab/class as example:

If uisng option '-builder' in swig : 

  swig  -scilab -builder  -buildersources example.cxx -c++  example.i

and generate loader.sce by exec builder.sce it generate following code inside  loader.sce 

      [bOK, ilib] = c_link('Class');

 For the case running swig without option   '-builder' 

   swig  -scilab   -buildersources example.cxx -c++  example.i

Swig generate loader.sce by itself with code like :

         [bOK, ilib] = c_link('libClass');


So, we should either using 
   saveBuilderFile(gatewayLibraryName);
   saveLoaderFile(gatewayLibraryName);
or 
   saveBuilderFile(gatewayName);
   saveLoaderFile(gatewayName)

in scilab.cxx.

Reply:
OK, but is one approach preferable to the other for any reason?